bfd_vrf_topo1 is failing for me on master

My branch:
sharpd@eva:~/frr4$ git status
On branch master
Your branch is up to date with 'origin/master'.

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	log.1

nothing added to commit but untracked files present (use "git add" to track)
sharpd@eva:~/frr4$ git show
commit 9b222c1f1a9bb31c8d186505a40158a145c97649 (HEAD -> master, origin/master, origin/HEAD)
Merge: 0d55f01bf bac31cb88
Author: Philippe Guibert <philippe.guibert@6wind.com>
Date:   Wed Jun 17 09:09:02 2020 +0200

    Merge pull request #6577 from rtrlib/2020-06-12-master-fixes
    
    bgpd: rpki fixes (master)

my configure line:
./configure --build=x86_64-linux-gnu --prefix=/usr --includedir=${prefix}/include --mandir=${prefix}/share/man --infodir=${prefix}/share/info --sys
confdir=/etc --libexecdir=${prefix}/lib/frr --enable-exampledir=/usr/share/doc/frr/examples/ --localstatedir=/var/run/frr --sbindir=/usr/lib/frr --sysc
onfdir=/etc/frr --enable-ospfapi=yes --enable-multipath=128 --enable-user=frr --enable-group=frr --enable-vty-group=frrvty --enable-configfile-mask=064
0 --enable-logfile-mask=0640 --enable-rtadv --with-libpam --enable-systemd=yes --enable-cumulus=yes --enable-vtysh=yes --with-vtysh-pager=cat --enable-
bgp-vnc=yes --enable-sharpd=yes --enable-realms --enable-dev-build --enable-protobuf --enable-snmp --enable-config-rollbacks --enable-zeromq --enable-f
pm --enable-cumulus --enable-rpki --enable-bfdd --enable-irdp=yes --enable-protobuf=yes --enable-werror build_alias=x86_64-linux-gnu --no-create --no-r
ecursion

running bfd-vrf-topo1 against this fails:
============================================================================= FAILURES =============================================================================
_______________________________________________________________________ test_bfd_connection ________________________________________________________________________

    def test_bfd_connection():
        "Assert that the BFD peers can find themselves."
        tgen = get_topogen()
        if tgen.routers_have_failure():
            pytest.skip(tgen.errors)
    
        logger.info("waiting for bfd peers to go up")
        for router in tgen.routers().values():
            json_file = "{}/{}/peers.json".format(CWD, router.name)
            expected = json.loads(open(json_file).read())
    
            test_func = partial(
                topotest.router_json_cmp, router, "show bfd peers json", expected
            )
            _, result = topotest.run_and_expect(test_func, None, count=8, wait=0.5)
            assertmsg = '"{}" JSON output mismatches'.format(router.name)
>           assert result is None, assertmsg
E           AssertionError: "r1" JSON output mismatches
E           assert json value is different (
E             --- Expected value
E             +++ Current value
E             @@ -2,0 +3,6 @@
E             +        "detect-multiplier": 3, 
E             +        "diagnostic": "neighbor signaled session down", 
E             +        "downtime": 0, 
E             +        "echo-interval": 50, ...
E             
E             ...Full output truncated (14 lines hidden), use '-vv' to show

test_bfd_vrf_topo1.py:181: AssertionError
=============================================================== 1 failed, 5 skipped in 16.37 seconds ===============================================================


I git bisected to this commit:
sharpd@eva:~/frr4$ git bisect bad
29191d31832717b7747ee777a500d3b964a35a1a is the first bad commit
commit 29191d31832717b7747ee777a500d3b964a35a1a
Author: Rafael Zalamena <rzalamena@opensourcerouting.org>
Date:   Tue May 12 16:38:33 2020 -0300
    bfdd: tell peer when our control plane is dead
    Simplify and fix the code that handles session teardown on control
    plane shutdown (either failure or graceful).
    - Don't move the `NULL` check inside `free` functions that expect
      data: it creates harder to understand flows.
    - Add some new debug messages to aid visualizing session deletions.
    - Add sanity check error message (if it ever happens).
    Signed-off-by: Rafael Zalamena <rzalamena@opensourcerouting.org>
 bfdd/ptm_adapter.c | 63 +++++++++++++++++++++++++++++++++++++++++++-----------
 1 file changed, 51 insertions(+), 12 deletions(-)

--- a/bgpd/bgp_bfd.c
+++ b/bgpd/bgp_bfd.c
@@ -207,15 +207,9 @@ static void bgp_bfd_update_peer(struct peer *peer)
  */
 void bgp_bfd_reset_peer(struct peer *peer)
 {
-	struct bfd_info *bfd_info;
-
 	if (!peer->bfd_info)
 		return;
-	bfd_info = (struct bfd_info *)peer->bfd_info;
 
-	/* if status is not down, reset bfd */
-	if (bfd_info->status != BFD_STATUS_DOWN)
-		bgp_bfd_peer_sendmsg(peer, ZEBRA_BFD_DEST_DEREGISTER);
 	bgp_bfd_peer_sendmsg(peer, ZEBRA_BFD_DEST_REGISTER);
 }
 

--- a/tests/topotests/bfd-vrf-topo1/test_bfd_vrf_topo1.py
+++ b/tests/topotests/bfd-vrf-topo1/test_bfd_vrf_topo1.py
@@ -129,14 +129,6 @@ def setup_module(mod):
     # Initialize all routers.
     tgen.start_router()
 
-    # Verify that we are using the proper version and that the BFD
-    # daemon exists.
-    for router in router_list.values():
-        # Check for Version
-        if router.has_version("<", "5.1"):
-            tgen.set_error("Unsupported FRR version")
-            break
-
 
 def teardown_module(_mod):
     "Teardown the pytest environment"

--- a/tests/topotests/bfd-vrf-topo1/test_bfd_vrf_topo1.py
+++ b/tests/topotests/bfd-vrf-topo1/test_bfd_vrf_topo1.py
@@ -168,7 +168,7 @@ def test_bfd_connection():
         test_func = partial(
             topotest.router_json_cmp, router, "show bfd peers json", expected
         )
-        _, result = topotest.run_and_expect(test_func, None, count=8, wait=0.5)
+        _, result = topotest.run_and_expect(test_func, None, count=16, wait=1)
         assertmsg = '"{}" JSON output mismatches'.format(router.name)
         assert result is None, assertmsg
 
@@ -212,7 +212,7 @@ def test_bgp_fast_convergence():
             "show ip bgp vrf {}-cust1 json".format(router.name),
             expected,
         )
-        _, res = topotest.run_and_expect(test_func, None, count=40, wait=0.5)
+        _, res = topotest.run_and_expect(test_func, None, count=40, wait=1)
         assertmsg = "{}: bgp did not converge".format(router.name)
         assert res is None, assertmsg
 
@@ -254,7 +254,7 @@ def test_bfd_fast_convergence():
         test_func = partial(
             topotest.router_json_cmp, router, "show bfd peers json", expected
         )
-        _, res = topotest.run_and_expect(test_func, None, count=20, wait=0.5)
+        _, res = topotest.run_and_expect(test_func, None, count=40, wait=1)
         assertmsg = '"{}" JSON output mismatches'.format(router.name)
         assert res is None, assertmsg
 
@@ -287,7 +287,7 @@ def test_bgp_fast_reconvergence():
             "show ip bgp vrf {}-cust1 json".format(router.name),
             expected,
         )
-        _, res = topotest.run_and_expect(test_func, None, count=3, wait=1)
+        _, res = topotest.run_and_expect(test_func, None, count=16, wait=1)
         assertmsg = "{}: bgp did not converge".format(router.name)
         assert res is None, assertmsg
 

